crates/fluvio-benchmark/src/stats_collector.rs

```diff
@@ -41,98 +37,33 @@ impl ProducerStat {
-        stats_sender: Sender<Stats>,
-        stats: Arc<AtomicStats>,
-    ) -> Result<(), std::io::Error> {
-        spawn(async move {
-            loop {
-                let stats_sender = stats_sender.clone();
-                let stats = Arc::clone(&stats);
-                let old_record_send = stats.record_send.load(std::sync::atomic::Ordering::Relaxed);
-                let old_record_bytes = stats
-                    .record_bytes
-                    .load(std::sync::atomic::Ordering::Relaxed);
-                sleep(Duration::from_secs(1)).await;
-                let first_start_time = stats.first_start_time.read().await;
-                if first_start_time.is_none() {
-                    continue;
-                }
-
-                let new_record_send = stats.record_send.load(std::sync::atomic::Ordering::Relaxed);
-                let new_record_bytes = stats
-                    .record_bytes
-                    .load(std::sync::atomic::Ordering::Relaxed);
-
-                if new_record_send == old_record_send {
-                    continue;
-                }
-
-                let record_send = new_record_send - old_record_send;
-                let record_bytes = new_record_bytes - old_record_bytes;
-                let elapsed = first_start_time.expect("start time").elapsed();
-                let elapsed_seconds = elapsed.as_millis() as f64 / 1000.0;
-                let records_per_sec = (new_record_send as f64 / elapsed_seconds).round() as u64;
-                let bytes_per_sec = (new_record_bytes as f64 / elapsed_seconds).round() as u64;
-
-                let hist = histogram.lock().await;
-                let latency_avg = hist.mean() as u64;
-                let latency_max = hist.value_at_quantile(1.0);
-
-                stats_sender
-                    .send(Stats {
-                        record_send,
-                        record_bytes,
-                        records_per_sec,
-                        bytes_per_sec,
-                        latency_avg,
-                        latency_max,
-                    })
-                    .await
-                    .expect("send stats");
-            }
-        });
-        Ok(())
-    }
-
     fn track_latency(
         num_records: u64,
         end_sender: Sender<EndProducerStat>,
-        histogram: Arc<Mutex<Histogram<u64>>>,
+        latencies: Arc<RwLock<Vec<u64>>>,
         event_receiver: Receiver<ProduceCompletionBatchEvent>,
-        stats: Arc<AtomicStats>,
+        total_stats: Arc<TotalStats>,
     ) {
         spawn(async move {
-            let hist = histogram.clone();
+            let latency_histogram = latencies.clone();
             while let Ok(event) = event_receiver.recv().await {
-                let hist = hist.clone();
-                let stats = stats.clone();
+                let latencies = latency_histogram.clone();
+                let stats = total_stats.clone();
                 spawn(async move {
```

Comment:
double spawn loop is inefficient.   it may contribute to some latency.  not show stopper for this PR, but should flatten this.  it's probably better perform accumulator with atomic counter callback then perform analytics in the stat collector